community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/conf.py

```diff
@@ -482,18 +482,24 @@ def gen_topology(lkp: util.Lookup) -> TopologyBuilder:
     return bldr
 
 
-def gen_topology_conf(lkp: util.Lookup) -> None:
-    """generate slurm topology.conf from config.yaml"""
+def gen_topology_conf(lkp: util.Lookup) -> bool:
+    """
+    Generates slurm topology.conf.
+    Returns whether the topology.conf got updated.
+    """
     bldr = gen_topology(lkp).compress()
     conf_file = Path(lkp.cfg.output_dir or slurmdirs.etc) / "cloud_topology.conf"
+    old_hash = util.hash_file(conf_file) if conf_file.exists() else ""
+
     with open(conf_file, "w") as f:
         f.writelines(FILE_PREAMBLE + "\n")
         for line in bldr.render_conf_lines():
             f.write(line)
             f.write("\n")
         f.write("\n")
-    util.chown_slurm(conf_file, mode=0o600)
```

Comment:
Why is this removed? Seems unrelated.

Reply:
`install_topology_conf` does it already. No need to duplicate.